src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -478,20 +478,20 @@ public void InitializeInstrumentationKey(ITelemetry telemetry)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public void Initialize(ITelemetry telemetry)
         {
-            string instrumentationKey = this.Context.InstrumentationKey;
-
-            if (string.IsNullOrEmpty(instrumentationKey))
-            {
-                instrumentationKey = this.configuration.InstrumentationKey;
-            }
-
             ISupportAdvancedSampling telemetryWithSampling = telemetry as ISupportAdvancedSampling;
 
             // Telemetry can be already sampled out if that decision was made before calling Track()
             bool sampledOut = telemetryWithSampling?.IsSampledOutAtHead ?? false;
 
             if (!sampledOut)
             {
+                string instrumentationKey = this.Context.InstrumentationKey;
```

Comment:
nit: Consider moving this down to line 523 where the instrumentationKey is being used.

Reply:
agree